mysql-test/suite/versioning/t/partition.test

```diff
@@ -332,6 +332,16 @@ create or replace trigger tr after insert on t4 for each row insert into t2
   ( select x from t3 ) union ( select x from t1 );
 insert into t4 values (1);
 
+--echo # MDEV-14821 Assertion failure
+create or replace table t1 (a int, b int) with system versioning;
+insert into t1 values (1,10),(2,20),(3,30);
+update t1 set b = 0;
+alter table t1 partition by system_time interval 1 second ( partition p1 history, partition p2 history, partition pn current );
+--sleep 2
```

Comment:
Is it reproducible via `LIMIT`? I don't like 2 seconds delay in test.

Reply:
I failed to reproduce it with `limit`.

Reply:
Rebase on trunk and use this case:
```sql
create or replace table t1 (x int) with system versioning;
insert into t1 values (0), (1);
update t1 set x= x + 1;
alter table t1 partition by system_time limit 1 (
    partition p1 history,
    partition p2 history,
    partition pn current);
delete from t1 where x = 1;
delete from t1 where x = 2;
```
Merge when done.